NOnion/Utility/FSharpUtil.fs

```diff
@@ -60,3 +60,34 @@ module FSharpUtil =
                 // none of the jobs passed to Async.Choice returns None
                 return failwith "unreachable"
         }
+
+    let Retry<'E1, 'E2 when 'E1 :> Exception and 'E2 :> Exception>
+        (method: unit -> Async<unit>)
+        (maxRetryCount: int)
+        =
+        let rec retryLoop(tryNumber: int) =
+            async {
+                try
+                    do! method()
+                with
+                | :? 'E1
```

Comment:
are you sure you don't need the `as ex` thing here too?

Reply:
I don't think so, it won't compile even.

Reply:
what? then the behaviour is different for 'E1 and 'E2?